bigc/api_client.py

```diff
@@ -12,21 +11,51 @@
 MAX_V3_PAGE_SIZE = 250
 
 
+class RequestOptions(TypedDict, total=False):
+    params: dict[str, Any] | None
+    headers: dict[str, Any] | None
+    timeout: float | None
```

Comment:
Just from skimming, I don't think it really makes sense expose a `headers` argument on all the endpoints... same thing with `params`, not every endpoint actually accepts query params in BigCommerce.